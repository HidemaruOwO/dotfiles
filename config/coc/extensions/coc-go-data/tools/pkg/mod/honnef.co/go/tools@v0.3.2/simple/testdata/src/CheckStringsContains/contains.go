package pkg

import (
	"bytes"
	"strings"
)

func fn() {
	_ = strings.IndexRune("", 'x') > -1 //@ diag(` strings.ContainsRune`)
	_ = strings.IndexRune("", 'x') >= 0 //@ diag(` strings.ContainsRune`)
	_ = strings.IndexRune("", 'x') > 0
	_ = strings.IndexRune("", 'x') >= -1
	_ = strings.IndexRune("", 'x') != -1 //@ diag(` strings.ContainsRune`)
	_ = strings.IndexRune("", 'x') == -1 //@ diag(`!strings.ContainsRune`)
	_ = strings.IndexRune("", 'x') != 0
	_ = strings.IndexRune("", 'x') < 0 //@ diag(`!strings.ContainsRune`)

	_ = strings.IndexAny("", "") > -1 //@ diag(` strings.ContainsAny`)
	_ = strings.IndexAny("", "") >= 0 //@ diag(` strings.ContainsAny`)
	_ = strings.IndexAny("", "") > 0
	_ = strings.IndexAny("", "") >= -1
	_ = strings.IndexAny("", "") != -1 //@ diag(` strings.ContainsAny`)
	_ = strings.IndexAny("", "") == -1 //@ diag(`!strings.ContainsAny`)
	_ = strings.IndexAny("", "") != 0
	_ = strings.IndexAny("", "") < 0 //@ diag(`!strings.ContainsAny`)

	_ = strings.Index("", "") > -1 //@ diag(` strings.Contains`)
	_ = strings.Index("", "") >= 0 //@ diag(` strings.Contains`)
	_ = strings.Index("", "") > 0
	_ = strings.Index("", "") >= -1
	_ = strings.Index("", "") != -1 //@ diag(` strings.Contains`)
	_ = strings.Index("", "") == -1 //@ diag(`!strings.Contains`)
	_ = strings.Index("", "") != 0
	_ = strings.Index("", "") < 0 //@ diag(`!strings.Contains`)

	_ = bytes.IndexRune(nil, 'x') > -1 //@ diag(` bytes.ContainsRune`)
	_ = bytes.IndexAny(nil, "") > -1   //@ diag(` bytes.ContainsAny`)
	_ = bytes.Index(nil, nil) > -1     //@ diag(` bytes.Contains`)
}
