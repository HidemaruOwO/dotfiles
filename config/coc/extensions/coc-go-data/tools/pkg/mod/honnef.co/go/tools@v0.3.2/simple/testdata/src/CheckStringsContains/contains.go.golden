package pkg

import (
	"bytes"
	"strings"
)

func fn() {
	_ = strings.ContainsRune("", 'x') //@ diag(` strings.ContainsRune`)
	_ = strings.ContainsRune("", 'x') //@ diag(` strings.ContainsRune`)
	_ = strings.IndexRune("", 'x') > 0
	_ = strings.IndexRune("", 'x') >= -1
	_ = strings.ContainsRune("", 'x')  //@ diag(` strings.ContainsRune`)
	_ = !strings.ContainsRune("", 'x') //@ diag(`!strings.ContainsRune`)
	_ = strings.IndexRune("", 'x') != 0
	_ = !strings.ContainsRune("", 'x') //@ diag(`!strings.ContainsRune`)

	_ = strings.ContainsAny("", "") //@ diag(` strings.ContainsAny`)
	_ = strings.ContainsAny("", "") //@ diag(` strings.ContainsAny`)
	_ = strings.IndexAny("", "") > 0
	_ = strings.IndexAny("", "") >= -1
	_ = strings.ContainsAny("", "")  //@ diag(` strings.ContainsAny`)
	_ = !strings.ContainsAny("", "") //@ diag(`!strings.ContainsAny`)
	_ = strings.IndexAny("", "") != 0
	_ = !strings.ContainsAny("", "") //@ diag(`!strings.ContainsAny`)

	_ = strings.Contains("", "") //@ diag(` strings.Contains`)
	_ = strings.Contains("", "") //@ diag(` strings.Contains`)
	_ = strings.Index("", "") > 0
	_ = strings.Index("", "") >= -1
	_ = strings.Contains("", "")  //@ diag(` strings.Contains`)
	_ = !strings.Contains("", "") //@ diag(`!strings.Contains`)
	_ = strings.Index("", "") != 0
	_ = !strings.Contains("", "") //@ diag(`!strings.Contains`)

	_ = bytes.ContainsRune(nil, 'x') //@ diag(` bytes.ContainsRune`)
	_ = bytes.ContainsAny(nil, "")   //@ diag(` bytes.ContainsAny`)
	_ = bytes.Contains(nil, nil)     //@ diag(` bytes.Contains`)
}
